sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpPolicyProviders.java

```diff
@@ -36,32 +39,18 @@ public static void addBeforeRetryPolicies(List<HttpPipelinePolicy> policies) {
      * @param policies Policy list to append the policies.
      */
     public static void addAfterRetryPolicies(List<HttpPipelinePolicy> policies) {
-        addRetryPolicies(policies, () -> getPolicyProviders(false, AfterRetryPolicyProvider.class));
+        addPolices(policies, AFTER_PROVIDER);
     }
 
-    private static void addRetryPolicies(List<HttpPipelinePolicy> policies,
-        Supplier<Iterator<? extends HttpPolicyProvider>> policySupplier) {
-        Iterator<? extends HttpPolicyProvider> it = policySupplier.get();
-        while (it.hasNext()) {
-            HttpPolicyProvider policyProvider = it.next();
-            HttpPipelinePolicy policy = policyProvider.create();
+    private static void addPolices(List<HttpPipelinePolicy> policies,
+        Iterable<? extends HttpPolicyProvider> providers) {
+        for (HttpPolicyProvider provider : providers) {
+            HttpPipelinePolicy policy = provider.create();
             if (policy == null) {
-                throw new NullPointerException("HttpPipelinePolicy created with " + policyProvider.getClass()
-                    + " resulted in a null policy");
+                throw new NullPointerException(String.format(INVALID_POLICY, provider.getClass()));
```

Comment:
Log and throw the exception. Also, this should have been caught in checkstyle.

Reply:
Being used from a static method which conflicts with the logging guidelines, we may want to amend those to allow them in a static only class.